srcpkgs/speed-dreams/template

```diff
@@ -20,16 +20,11 @@ distfiles="${SOURCEFORGE_SITE}/speed-dreams/speed-dreams-src-base-${version}-${_
  ${SOURCEFORGE_SITE}/speed-dreams/speed-dreams-src-wip-cars-and-tracks-${version}-${_rev}.tar.xz
  ${SOURCEFORGE_SITE}/speed-dreams/speed-dreams-src-hq-cars-and-tracks-${version}-${_rev}.tar.xz
  ${SOURCEFORGE_SITE}/speed-dreams/speed-dreams-src-more-hq-cars-and-tracks-${version}-${_rev}.tar.xz"
-checksum="32aa55c17f8dafc1c54fb0d0b7b993d2688d431899cfc5db9eeece148c6887d0
- 9808d6d77b27cba2ca21943d1cff3c1f4022428cca4f76a2873eabe6954f4df1
- d8ebe92ebeb88bdf4339de9da5fa20da3d0873a6c2776f23f18e61fc6f510b51
- 0b7b417d007ab3b7f0f4dcfdc868aafee1722150b81af519e1cd4c082b408183"
+checksum="c67e48a083cd86f6c0062f0b38ed91a5d25274ea8aa656ee182f3f5074e96ff0
+ aff20e49ff94660d9ef962079f640d0ceeaa414869209be1ec41768d0ee5cb20
+ 06eaf2094bc10fbaca980a4236c4e444dd14275aba3f7b1d62b4b3a53a9d4c77
+ 19208761b252df9db200c733cecb3bb584905f02d057af2bbff5e4c86cbe1a86"
 nocross=yes
 replaces="speed-dreams-data>=0"
-CFLAGS=-fpermissive
-CXXFLAGS=-fpermissive
+LDFLAGS="-Wl,--no-as-needed"
```

Comment:
Can you add a comment explaining this? I suppose it depends on some library's constructor...

Reply:
> Can you add a comment explaining this? I suppose it depends on some library's constructor...

Honestly I don't know why `--as-needed` is not supported, I just know it makes the build fail.
I will report it to upstream. I see they had the same problem in the past and then it was fixed, this is a regression.

Reply:
Can you mention in the commit message as well, then? Or even add a comment on top, that removal should be attempted when updated.

Reply:
When cmake policy [CMP0072](https://cmake.org/cmake/help/v3.21/policy/CMP0072.html) is set to `NEW`, linking with `--as-needed` fails. Should I patch `CMakeLists.txt` to use the legacy library or just use `--no-as-needed`?

Reply:
Huh, I didn't expect that. Theoretically we should pass it for all Cmake builds, since they are eventually going to start using that policy...

Reply:
I don't know if other packages are affected.
Anyway, another workaround could be to pass `-lGL` only when linking the problematic files (I tested this and it works), but mixing legacy and new libraries is ugly.